packages/office-ui-fabric-react/src/components/ProgressIndicator/examples/ProgressIndicator.Basic.Example.tsx

```diff
@@ -1,65 +1,22 @@
 import * as React from 'react';
 import { ProgressIndicator } from 'office-ui-fabric-react/lib/ProgressIndicator';
-import { Async } from 'office-ui-fabric-react/lib/Utilities';
 
-export interface IProgressIndicatorBasicExampleState {
-  percentComplete: number;
-}
+const intervalDelay = 100;
+const intervalIncrement = 0.01;
 
-const INTERVAL_DELAY = 100;
-const INTERVAL_INCREMENT = 0.01;
-const RESTART_WAIT_TIME = 2000;
+export const ProgressIndicatorBasicExample: React.FunctionComponent = () => {
+  const [percentComplete, setPercentComplete] = React.useState(0);
 
-export class ProgressIndicatorBasicExample extends React.Component<{}, IProgressIndicatorBasicExampleState> {
-  private _interval: number;
-  private _async: Async;
-
-  constructor(props: {}) {
-    super(props);
-
-    this._async = new Async(this);
-
-    this.state = {
-      percentComplete: 0,
+  React.useEffect(() => {
+    const id = setInterval(() => {
+      setPercentComplete((intervalIncrement + percentComplete) % 1);
```

Comment:
You should add back logic to stop the interval after it reaches 100% :)

Reply:
I worked through this with him and had suggested that simplification. Do you think it's very important to keep?

Reply:
Progress over 100% might be ignored, but I still think it's cleaner to detect and stop when it's finished.